src/main/java/com/volmit/iris/engine/object/IrisEffect.java

```diff
@@ -151,6 +147,36 @@
     @MinNumber(1)
     @Desc("The chance is 1 in CHANCE per interval")
     private int chance = 50;
+    @ArrayType(min = 1, type = IrisCommand.class)
+    @Desc("Run commands, at the exact location of the player")
+    private KList<IrisCommand> rawCommands = new KList<>();
+    @DependsOn({"rawCommands"})
+    @MinNumber(-8)
+    @MaxNumber(8)
+    @Desc("The alt x, usually represents motion if the particle count is zero. Otherwise an offset.")
+    private double commandOffsetX = 0;
+    @DependsOn({"rawCommands"})
+    @MinNumber(-8)
+    @MaxNumber(8)
+    @Desc("The alt y, usually represents motion if the particle count is zero. Otherwise an offset.")
+    private double commandOffsetY = 0;
+    @DependsOn({"rawCommands"})
```

Comment:
Put all of these objects into a sub object such as 

@Desc("Configuration for a command")
class IrisCommandRegistry{
rawCommands 
commandOffsetX Y Z
...
}

Then simply define in this class the IrisCommandRegistry field as a sep config. Keeps stuff clean